lsp/src/main/java/org/aya/lsp/options/RenderOptions.java

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.lsp.options;
+
+import kala.collection.Seq;
+import kala.collection.immutable.ImmutableMap;
+import kala.collection.mutable.MutableMap;
+import kala.control.Option;
+import kala.control.Result;
+import org.aya.lsp.models.ServerOptions;
+import org.aya.pretty.backend.html.DocHtmlPrinter;
+import org.aya.pretty.backend.latex.DocTeXPrinter;
+import org.aya.pretty.doc.Doc;
+import org.aya.pretty.printer.ColorScheme;
+import org.aya.pretty.printer.StyleFamily;
+import org.aya.pretty.style.AyaColorScheme;
+import org.aya.pretty.style.AyaStyleFamily;
+import org.javacs.lsp.MarkupKind;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @param colorScheme
+ * @param styleFamily
+ */
+public record RenderOptions(
+  @NotNull Option<ColorScheme> colorScheme,
+  @NotNull Option<StyleFamily> styleFamily
+) {
+  public final static RenderOptions DEFAULT = new RenderOptions(
+    Option.some(AyaColorScheme.EMPTY),
+    Option.some(AyaStyleFamily.DEFAULT)
+  );
+
+  public enum RenderTarget {
+    Debug,
+    HTML,
+    TeX
+  }
+
+  public static final @NotNull ImmutableMap<String, ColorScheme> BUILTIN_COLOR_SCHEMES = ImmutableMap.of(
+    "emacs", AyaColorScheme.EMACS,
+    "intellij", AyaColorScheme.INTELLIJ,
+    "none", AyaColorScheme.EMPTY
+  );
+
+  /// region Helper
+
+  /**
+   * Construct a {@link RenderOptions} from {@link ServerOptions}
+   */
+  @Contract(pure = true)
+  public static @NotNull Result<@NotNull RenderOptions, @NotNull String> fromServerOptions(@NotNull ServerOptions options) {
+    Option<ColorScheme> colorScheme = Option.none();
+
+    // colorScheme is none if both colorName and colorOverride are null
+    if (options.colorName != null || options.colorOverride != null) {
+      Result<ColorScheme, String> parsed = parseColorScheme(options);
+      if (parsed.isErr()) return Result.err(parsed.getErr());
+
+      colorScheme = Option.some(parsed.get());
+    }
+    // TODO[hoshino]: styleFaimily
+
+
+    return Result.ok(new RenderOptions(colorScheme, Option.none()));
+  }
+
+  /**
+   * @apiNote this function returns {@code Result.ok(AyaColorScheme.EMPTY)} even though {@code options.colorName and options.colorOverride are null}
+   */
+  private static Result<ColorScheme, String> parseColorScheme(@NotNull ServerOptions options) {
+    var colorName = options.colorName;
+    var baseOption = options.colorName == null
+      ? Option.some(AyaColorScheme.EMPTY)
+      : BUILTIN_COLOR_SCHEMES.getOption(colorName);
+
+    if (baseOption.isEmpty()) {
+      return Result.err("Invalid color scheme name");
+    }
```

Comment:
When the client sets `colorName` to any invalid name, say `invalid`, we will throw this error instead of using `EMPTY`. Is this intentional? I suggest removing `none` from `BUILTIN_COLOR_SCHEMES` and using
```java
var baseOption = BUILTIN_COLOR_SCHEMES.getOrDefault(colorName, AyaColorScheme.EMPTY);
```

Because the empty color scheme should be a fallback rather than an option.